specs/Pills.md

```diff
@@ -0,0 +1,182 @@
+# [Pills]()
+
+## Background
+
+PIlls should be used when representing an input, as a way to filter content, or to represent an attribute.
+
+## Prior Art
+
+- [Evergreen](https://evergreen.segment.com/components/badge-and-pill/)
+- [Lightning](https://www.lightningdesignsystem.com/components/pills/)
+- [Material UI](https://material-ui.com/components/chips/#chip)
+- [OpenUI Research](https://github.com/WICG/open-ui/pull/259)
+
+## Sample Code
+
+```jsx
+<Pill>Item 1</Pill>
+<Pill clickable>Item 2</Pill>
+<TogglePill>On</Pill>
+```
+
+## Variants
+
+- Shape: `circular`, `rounded`
+- Size: `smaller`, `small`, `medium`
+- Appearance: `Filled Grey (filled)`, `Filled White (inverted)`, `Outline`
+
+## PROPS
+
+```typescript
+export type PillShape = 'rounded' | 'circular';
+
+export type PillAppearence = 'filled' | 'inverted' | 'outline';
+
+export type SizeValue = 'smaller' | 'small' | 'medium';
+
+export interface PillProps extends ComponentProps, React.HTMLAttributes<HTMLElement> {
+  /**
+   * A Pill can be sized.
+   */
+  size?: SizeValue;
+
+  /**
+   * A Pill can be circular or rounded
+   */
+  shape?: PillShape;
+
+  /**
+   * A Pill can be filled, inverted or outline
+   */
+  appearance?: PillVariant;
+
+  /**
+   * Media slot
+   */
+  media?: ShorthandProps<HTMLElement>;
+
+  /**
+   * Details that will appear as second line
+   */
+  details?: ShorthandProps<HTMLElement>;
+
+  /** A Pill can be dismissible. */
+  dismissible?: boolean;
+
+  /** A Pill can be disabled. */
+  disabled?: boolean;
+
+  /** A Pill can be selected. */
+  selected?: boolean;
+
+  /** A Pill can be clickable */
+  clickable?: boolean;
+
+  /** A callback to be called when Pill is clicked */
+  onClick?: function;
+
+  /**
+   * A button shorthand for the dismiss action slot. To use this slot the pill should be
+   * dismissible.
+   */
+  dismissIcon?: ShorthandProps<HTMLElement>;
+}
+
+export type TogglePillProps = Omit<PillProps, 'clickable'>;
+```
+
+## Structure
+
+- _**Public**_
+
+```jsx
+<Pill />
+
+<Pill clickable />
+
+<TogglePill />
+```
+
+- _**DOM**_
+
+```html
+<span>
+  ...
+</span>
+
+<span role="button">
+  ...
+</span>
+
+<span role="switch" aria-checked>
+  ...
+</span>
+```
+
+## Migration
+
+- _Migration from v8_
+
+There's no components related to `Pill` in V8
+
+- _Migration from v0_
+
+`Dropdown` component should use `Pills` over `DropdownSelectedItem`.
+
+### Pill with dismiss
+
+#### Accessibility
+
+- Tab key to set focus on the Pill
```

Comment:
during the design review, listbox/option were proposed - please check also the pills in dropdown. With that, arrow key navigation would be used to navigate. We should even do left+up arrow to move to previous, right+down arrow to move to next.

Reply:
I also had conversation with @kolaps33 about it, can you express your opinions  again @kolaps33

Reply:
we should not rely on using plain divs to form appropriate strucutre. We need to have `Pills` component that defines keyboard navigation and roles properly. Modifications can be done using behaviors, not primarily using role overrides.

In most cases, we want to reduce the amount of tab stops - that is typically achieved by using arrow keys in places where possible.

Reply:
Updated to: 

`- Arrow keys navigates between `Pill``